mkosi/config.py

```diff
@@ -1928,10 +1937,8 @@ def match_config(path: Path, namespace: argparse.Namespace, defaults: argparse.N
             if not v:
                 die("Match value cannot be empty")
 
-            if (s := settings_lookup_by_name.get(k)):
-                if not s.match:
-                    die(f"{k} cannot be used in [Match]")
-
+            if (s := settings_lookup_by_name.get((section, k))):
+                assert s.match
```

Comment:
Why downgrade this to an `assert`? `assert`s can be removed from Python code easily, just by setting an environment variable. Preferably `assert`s shouldn't be used outside of tests in most cases.

Reply:
That's the point actually, the `assert` is only there to let the type checker know that `s.match` is not `None`.

We know it's not `None` because we are only parsing the `[Match]` section here, so `section == "Match"`. All the keys in `settings_lookup_by_name` with a `"Match"` section are there because the setting has a `match` function (see `iter_sections()`).

Reply:
I restored the old code when downgrading to warnings. And even if we turn it back in a hard error, the fuzzy lookup might still be useful for the message.